soc/arm/nordic_nrf/nrf52/dts_fixup.h

```diff
@@ -2,166 +2,30 @@
 
 #define DT_NUM_IRQ_PRIO_BITS	DT_ARM_V7M_NVIC_E000E100_ARM_NUM_IRQ_PRIORITY_BITS
 
-#define DT_ADC_0_IRQ		DT_NORDIC_NRF_SAADC_40007000_IRQ_0
-#define CONFIG_ADC_0_IRQ_PRI		DT_NORDIC_NRF_SAADC_40007000_IRQ_0_PRIORITY
-#define CONFIG_ADC_0_NAME		DT_NORDIC_NRF_SAADC_40007000_LABEL
+#define CONFIG_ADC_0_NAME		DT_NORDIC_NRF_SAADC_ADC_0_LABEL
 
-#if defined(DT_NORDIC_NRF_UARTE_40002000_BASE_ADDRESS)
-#define DT_UART_0_BASE		DT_NORDIC_NRF_UARTE_40002000_BASE_ADDRESS
-#define DT_UART_0_IRQ_PRI		DT_NORDIC_NRF_UARTE_40002000_IRQ_0_PRIORITY
-#define DT_UART_0_IRQ_NUM		DT_NORDIC_NRF_UARTE_40002000_IRQ_0
-#define DT_UART_0_BAUD_RATE		DT_NORDIC_NRF_UARTE_40002000_CURRENT_SPEED
-#define DT_UART_0_NAME		DT_NORDIC_NRF_UARTE_40002000_LABEL
-#define DT_UART_0_TX_PIN		DT_NORDIC_NRF_UARTE_40002000_TX_PIN
-#define DT_UART_0_RX_PIN		DT_NORDIC_NRF_UARTE_40002000_RX_PIN
-  #if defined(DT_NORDIC_NRF_UARTE_40002000_RTS_PIN)
-  #define DT_UART_0_RTS_PIN		DT_NORDIC_NRF_UARTE_40002000_RTS_PIN
-  #endif
-  #if defined(DT_NORDIC_NRF_UARTE_40002000_CTS_PIN)
-  #define DT_UART_0_CTS_PIN		DT_NORDIC_NRF_UARTE_40002000_CTS_PIN
-  #endif
-#else
-#define DT_UART_0_BASE		DT_NORDIC_NRF_UART_40002000_BASE_ADDRESS
-#define DT_UART_0_IRQ_PRI		DT_NORDIC_NRF_UART_40002000_IRQ_0_PRIORITY
-#define DT_UART_0_IRQ_NUM		DT_NORDIC_NRF_UART_40002000_IRQ_0
-#define DT_UART_0_BAUD_RATE		DT_NORDIC_NRF_UART_40002000_CURRENT_SPEED
-#define DT_UART_0_NAME		DT_NORDIC_NRF_UART_40002000_LABEL
-#define DT_UART_0_TX_PIN		DT_NORDIC_NRF_UART_40002000_TX_PIN
-#define DT_UART_0_RX_PIN		DT_NORDIC_NRF_UART_40002000_RX_PIN
-  #if defined(DT_NORDIC_NRF_UART_40002000_RTS_PIN)
-  #define DT_UART_0_RTS_PIN		DT_NORDIC_NRF_UART_40002000_RTS_PIN
-  #endif
-  #if defined(DT_NORDIC_NRF_UART_40002000_RTS_PIN)
-  #define DT_UART_0_CTS_PIN		DT_NORDIC_NRF_UART_40002000_CTS_PIN
-  #endif
-#endif
-
-#define DT_UART_1_BASE		DT_NORDIC_NRF_UARTE_40028000_BASE_ADDRESS
-#define DT_UART_1_IRQ_PRI		DT_NORDIC_NRF_UARTE_40028000_IRQ_0_PRIORITY
-#define DT_UART_1_IRQ_NUM		DT_NORDIC_NRF_UARTE_40028000_IRQ_0
-#define DT_UART_1_BAUD_RATE		DT_NORDIC_NRF_UARTE_40028000_CURRENT_SPEED
-#define DT_UART_1_NAME		DT_NORDIC_NRF_UARTE_40028000_LABEL
-#define DT_UART_1_TX_PIN		DT_NORDIC_NRF_UARTE_40028000_TX_PIN
-#define DT_UART_1_RX_PIN		DT_NORDIC_NRF_UARTE_40028000_RX_PIN
-#if defined(DT_NORDIC_NRF_UARTE_40028000_RTS_PIN)
-#define DT_UART_1_RTS_PIN		DT_NORDIC_NRF_UARTE_40028000_RTS_PIN
-#endif
-#if defined(DT_NORDIC_NRF_UARTE_40028000_CTS_PIN)
-#define DT_UART_1_CTS_PIN		DT_NORDIC_NRF_UARTE_40028000_CTS_PIN
-#endif
+#define CONFIG_UART_0_NAME		DT_NORDIC_NRF_UART_UART_0_LABEL
+#define CONFIG_UART_1_NAME		DT_NORDIC_NRF_UART_UART_1_LABEL
 
 #define DT_FLASH_DEV_NAME			DT_NRF_NRF52_FLASH_CONTROLLER_4001E000_LABEL
 
-#define DT_GPIO_P0_DEV_NAME		DT_NORDIC_NRF_GPIO_50000000_LABEL
+#define DT_GPIO_P0_DEV_NAME		DT_NORDIC_NRF_GPIO_GPIO_0_LABEL
 #if CONFIG_HAS_HW_NRF_GPIO1
```

Comment:
I think we don't need this dependency on a Kconfig symbol here. `DT_GPIO_P1_DEV_NAME` can be defined without any conditions, just like `DT_GPIO_P0_DEV_NAME` is.